# COMPLETE TESS LIGHT CURVE INFORMATION

This notebook takes the GTO objects list and tries to first simply resolve the ambiguity, taking the first file in folder. If not, try to complete the TESS light curve information.

## Modules and configuration

### Modules

In [1]:
# Modules import:
import pandas as pd
import numpy as np
import os
from time import sleep

#from astropy.table import Table, QTable
#from astropy.timeseries import TimeSeries
#from astropy import units as u
#from astropy.time import Time
from astropy.io import fits

import lightkurve as lk

#%matplotlib inline
#import matplotlib.pyplot as plt

#from pylab import rcParams
#rcParams['figure.figsize'] = 11, 11

#import seaborn as sns
#sns.set_style("white", {'figure.figsize':(15,10)})
#sns.set_style("whitegrid")
#sns.set(rc={'figure.figsize':(15,8)})

### Configuration

In [2]:
# Configuration:
DATA_FILE = "../data/CARM_VIS_objects_with_PG.csv"
TESS_LC_FOLDER = "../data/CARMENES_GTO_TESS_lc/lightcurves/"


## TESS file correction

In [3]:
gto = pd.read_csv(DATA_FILE, sep=',', decimal='.')
gto.head()

Karmn             Name Comp    GJ  RA_J2016_deg  DE_J2016_deg  \
0  J23585+076        Wolf 1051   AB  4383    359.636420      7.656947   
1  J23556-061           GJ 912   AB   912    358.913617     -6.144283   
2  J23548+385  RX J2354.8+3831    -   NaN    358.713658     38.526340   
3  J23505-095       LP 763-012    -  4367    357.634705     -9.560964   
4  J23492+024           BR Psc    -   908    357.306604      2.396918   

      RA_J2000     DE_J2000  l_J2016_deg  b_J2016_deg  ...   T0_PG_TESS  \
0  23:58:32.65  +07:39:30.1   100.839686   -52.931049  ...          NaN   
1  23:55:39.78  -06:08:33.4    88.129933   -65.175491  ...          NaN   
2  23:54:51.46  +38:31:36.2   110.941908   -23.024449  ...  1764.609498   
3  23:50:31.64  -09:33:32.7    80.777067   -67.303426  ...  1354.108815   
4  23:49:12.53  +02:24:04.4    93.567467   -56.885396  ...          NaN   

  e_T0_PG_TESS  offset_PG_TESS e_offset_PG_TESS  FAP_PG_TESS  valid_PG_TESS  \
0          NaN             NaN              NaN          NaN            0.0   
1          NaN             NaN              NaN          NaN            0.0   
2     0.010704     8262.219751         1.365667     1.000000            1.0   
3     0.001261     7767.134654         0.094298     0.064148            1.0   
4          NaN             NaN              NaN          NaN            0.0   

                                       error_PG_TESS  elapsed_time_PG_TESS  \
0  Not recognized as a supported data product:\nn...              0.002992   
1  Not recognized as a supported data product:\nn...              0.001671   
2                                                NaN            344.002685   
3                                                NaN            473.533042   
4  Not recognized as a supported data product:\nn...              0.000997   

                                      fits_file_TESS  \
0                                                NaN   
1                                                NaN   
2  ../data/CARMENES_GTO_TESS_PGs/J23548+385_TESS_...   
3  ../data/CARMENES_GTO_TESS_PGs/J23505-095_TESS_...   
4                                                NaN   

                                       fig_file_TESS  
0                                                NaN  
1                                                NaN  
2  ../data/CARMENES_GTO_TESS_PGs/figures/J23548+3...  
3  ../data/CARMENES_GTO_TESS_PGs/figures/J23505-0...  
4                                                NaN  

[5 rows x 248 columns]

In [4]:
gto['has_lc']

0      False
1      False
2       True
3       True
4      False
       ...  
385     True
386    False
387    False
388    False
389     True
Name: has_lc, Length: 390, dtype: bool

In [5]:
gto['lc_file']

0                                                    NaN
1                                                    NaN
2      ../data/CARMENES_GTO_TESS_lc/lightcurves/J2354...
3      ../data/CARMENES_GTO_TESS_lc/lightcurves/J2350...
4                                                    NaN
                             ...                        
385    ../data/CARMENES_GTO_TESS_lc/lightcurves/J0018...
386                                                  NaN
387                                                  NaN
388                                                  NaN
389    ../data/CARMENES_GTO_TESS_lc/lightcurves/J0005...
Name: lc_file, Length: 390, dtype: object

In [6]:
print(list(gto.columns))

['Karmn', 'Name', 'Comp', 'GJ', 'RA_J2016_deg', 'DE_J2016_deg', 'RA_J2000', 'DE_J2000', 'l_J2016_deg', 'b_J2016_deg', 'Ref01', 'SpT', 'SpTnum', 'Ref02', 'Teff_K', 'eTeff_K', 'logg', 'elogg', '[Fe/H]', 'e[Fe/H]', 'Ref03', 'L_Lsol', 'eL_Lsol', 'Ref04', 'R_Rsol', 'eR_Rsol', 'Ref05', 'M_Msol', 'eM_Msol', 'Ref06', 'muRA_masa-1', 'emuRA_masa-1', 'muDE_masa-1', 'emuDE_masa-1', 'Ref07', 'pi_mas', 'epi_mas', 'Ref08', 'd_pc', 'ed_pc', 'Ref09', 'Vr_kms-1', 'eVr_kms-1', 'Ref10', 'ruwe', 'Ref11', 'U_kms-1', 'eU_kms-1', 'V_kms-1', 'eV_kms-1', 'W_kms-1', 'eW_kms-1', 'Ref12', 'sa_m/s/a', 'esa_m/s/a', 'Ref13', 'SKG', 'Ref14', 'SKG_lit', 'Ref14_lit', 'Pop', 'Ref15', 'vsini_flag', 'vsini_kms-1', 'evsini_kms-1', 'Ref16', 'P_d', 'eP_d', 'Ref17', 'pEWHalpha_A', 'epEWHalpha_A', 'Ref18', 'log(LHalpha/Lbol)', 'elog(LHalpha/Lbol)', 'Ref19', '1RXS', 'CRT_s-1', 'eCRT_s-1', 'HR1', 'eHR1', 'HR2', 'eHR2', 'Flux_X_E-13_ergcm-2s-1', 'eFlux_X_E-13_ergcm-2s-1', 'LX/LJ', 'eLX/LJ', 'Ref20', 'Activity', 'Ref21', 'FUV_mag',

In [7]:
type(gto.loc[0, 'TIC_id'])

numpy.int64

In [8]:
#for i in range (0,20):
for i in range (136, len(gto)):
    filename = gto.loc[i, 'lc_file']
    # See if it exists:
    try:
        karmn_name = gto.loc[i, 'Karmn']
        object_name = gto.loc[i, 'Name']
        tic_id = "TIC" + str(gto.loc[i, 'TIC_id'])
        target_folder = TESS_LC_FOLDER + karmn_name + "/"
        if filename == "<ambiguous>":
            # Select the first file in folder:
            files = os.listdir(target_folder)
            selected_file = files[0]
            print("<ambiguous>: selecting the first file in %s folder: %s" 
                  %(target_folder, selected_file))
            gto.loc[i, 'lc_file'] = target_folder + selected_file
            gto.loc[i, 'has_lc'] = True
        elif filename is np.nan or not os.path.exists(filename):
            sleep(30)
            print("%s filename DOES NOT exist, trying to look and download" %filename)
            lk.search.log.setLevel("DEBUG")
            #search_result = lk.search_lightcurve(object_name, mission='TESS', exptime=600)
            search_result = lk.search_lightcurve(tic_id, mission='TESS', exptime=1800)
            if len(search_result) > 0:
                # Select the item with greater exposition time, download it and remove NaN values:
                exp_times = np.array(search_result.exptime.value)
                max_idx = np.argmax(exp_times)
                lc = search_result[max_idx].download().remove_nans()
                sector = search_result[max_idx].mission[0][-2:]
                # Store it and update the info in the file:
                if not os.path.exists(target_folder):
                    os.makedirs(target_folder)
                else:
                    pass
                target_filename = target_folder + \
                    karmn_name + "_TESS_" + tic_id + "_S" + sector + ".fits"
                lc.to_fits(path=target_filename, overwrite=True, flux_column_name='flux')
                gto.loc[i, 'lc_file'] = target_filename
                gto.loc[i, 'has_lc'] = True
                print("...SUCCESS: found object and stored file %s" %target_filename)
            else:
                # No results returned
                print("** WARNING: no results found for %s star" %karmn_name)
        else:
            print("%s filename DOES exist" %filename)
    except Exception as e:
            print("*** SOME ERROR happened. Error = %s" %str(e))
        


../data/CARMENES_GTO_TESS_lc/lightcurves/J15218+209/J15218+209_TESS_TIC355793860_S24.fits filename DOES exist


Started querying MAST for observations with the exact target_name='036853511'.


nan filename DOES NOT exist, trying to look and download


No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC36853511'.
MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC36853511".


** WARNING: no results found for J15194-077 star


Started querying MAST for observations with the exact target_name='351346161'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC351346161'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC351346161".


** WARNING: no results found for J15100+193 star


Started querying MAST for observations with the exact target_name='460418429'.
No observations found. Now performing a cone search instead.


nan filename DOES NOT exist, trying to look and download


Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC460418429'.
MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC460418429".


** WARNING: no results found for J15095+031 star


Started querying MAST for observations with the exact target_name='460356719'.
No observations found. Now performing a cone search instead.


nan filename DOES NOT exist, trying to look and download


Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC460356719'.
MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC460356719".


** WARNING: no results found for J15013+055 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J14544+355/J14544+355_TESS_TIC284700100_S23.fits filename DOES exist


Started querying MAST for observations with the exact target_name='229550884'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC229550884'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC229550884".


** WARNING: no results found for J14524+123 star


Started querying MAST for observations with the exact target_name='294401924'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC294401924'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC294401924".


** WARNING: no results found for J14342-125 star


Started querying MAST for observations with the exact target_name='349568452'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC349568452'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC349568452".


** WARNING: no results found for J14321+081 star


Started querying MAST for observations with the exact target_name='297175954'.
No observations found. Now performing a cone search instead.


nan filename DOES NOT exist, trying to look and download


Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC297175954'.
MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC297175954".


** WARNING: no results found for J14310-122 star


Started querying MAST for observations with the exact target_name='023741668'.
No observations found. Now performing a cone search instead.


nan filename DOES NOT exist, trying to look and download


Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC23741668'.
MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC23741668".


** WARNING: no results found for J14307-086 star


Started querying MAST for observations with the exact target_name='450352108'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J14294+155 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J14257+236W/J14257+236W_TESS_TIC298434280_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J14257+236E/J14257+236E_TESS_TIC298434278_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J14251+518/J14251+518_TESS_TIC441709023_S16.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J14173+454/J14173+454_TESS_TIC168747450_S16.fits filename DOES exist


Started querying MAST for observations with the exact target_name='446416116'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J14155+046 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J14152+450/J14152+450_TESS_TIC168707044_S16.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J14082+805/J14082+805_TESS_TIC310000679_S14.fits filename DOES exist


Started querying MAST for observations with the exact target_name='119147875'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J14010-026 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J13591-198/J13591-198_TESS_TIC125421413_S11.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J13582+125/J13582+125_TESS_TIC387009593_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J13536+776/J13536+776_TESS_TIC219463771_S14.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J13458-179/J13458-179_TESS_TIC286942673_S37.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J13457+148/J13457+148_TESS_TIC72465347_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J13450+176/J13450+176_TESS_TIC72464819_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J13427+332/J13427+332_TESS_TIC173095595_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J13299+102/J13299+102_TESS_TIC404519959_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J13293+114/J13293+114_TESS_TIC40

Started querying MAST for observations with the exact target_name='115920948'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC115920948'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC115920948".


** WARNING: no results found for J12428+418 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J12388+116/J12388+116_TESS_TIC399119319_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J12373-208/J12373-208_TESS_TIC434508218_S10.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J12350+098/J12350+098_TESS_TIC399089737_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J12312+086/J12312+086_TESS_TIC399057352_S23.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J12248-182/J12248-182_TESS_TIC22395594_S10.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J12230+640/J12230+640_TESS_TIC147670988_S14.fits filename DOES exist


Started querying MAST for observations with the exact target_name='389356212'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J12189+111 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J12156+526/J12156+526_TESS_TIC416538839_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J12123+544S/J12123+544S_TESS_TIC224289449_S15.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J12111-199/J12111-199_TESS_TIC422382572_S10.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J12100-150/J12100-150_TESS_TIC347694620_S10.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J12054+695/J12054+695_TESS_TIC142963210_S14.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11511+352/J11511+352_TESS_TIC144400022_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11509+483/J11509+483_TESS_TIC11893637_S22.fits filename DOES exist


Started querying MAST for observations with the exact target_name='325275315'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J11477+008 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J11476+786/J11476+786_TESS_TIC358615829_S14.fits filename DOES exist


Started querying MAST for observations with the exact target_name='325273691'.
MAST found 1 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J11476+002 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J11474+667/J11474+667_TESS_TIC142878895_S14.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11467-140/J11467-140_TESS_TIC386627655_S09.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11423+230/J11423+230_TESS_TIC119584412_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11421+267/J11421+267_TESS_TIC138819293_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11417+427/J11417+427_TESS_TIC115869504_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11306-080/J11306-080_TESS_TIC285972083_S09.fits filename DOES exist


Started querying MAST for observations with the exact target_name='388804061'.
MAST found 3 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J11302+076 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J11289+101/J11289+101_TESS_TIC388802555_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11201-104/J11201-104_TESS_TIC453465810_S09.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11126+189/J11126+189_TESS_TIC356668126_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11110+304W/J11110+304W_TESS_TIC9001924_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11110+304E/J11110+304E_TESS_TIC9001920_S22.fits filename DOES exist


Started querying MAST for observations with the exact target_name='252803606'.
MAST found 1 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 1 matching data products.
File found in local cache.


*** SOME ERROR happened. Error = object has no attribute centroid_col
../data/CARMENES_GTO_TESS_lc/lightcurves/J11054+435/J11054+435_TESS_TIC252803603_S21.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11044+304/J11044+304_TESS_TIC166648874_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11033+359/J11033+359_TESS_TIC166646191_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11026+219/J11026+219_TESS_TIC97488127_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J11000+228/J11000+228_TESS_TIC97472519_S22.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10584-107/J10584-107_TESS_TIC148563078_S09.fits filename DOES exist


Started querying MAST for observations with the exact target_name='365006789'.
MAST found 5 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J10564+070 star


Started querying MAST for observations with the exact target_name='393584005'.
MAST found 4 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J10508+068 star


Started querying MAST for observations with the exact target_name='165654322'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J10504+331 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J10482-113/J10482-113_TESS_TIC471015913_S09.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10416+376/J10416+376_TESS_TIC17299996_S21.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10396-069/J10396-069_TESS_TIC54959262_S09.fits filename DOES exist


Started querying MAST for observations with the exact target_name='374267666'.


nan filename DOES NOT exist, trying to look and download


MAST found 5 observations. Now querying MAST for the corresponding data products.
MAST found 0 matching data products.


** WARNING: no results found for J10360+051 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J10354+694/J10354+694_TESS_TIC103803629_S14.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10350-094/J10350-094_TESS_TIC56665956_S09.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10289+008/J10289+008_TESS_TIC277888583_S35.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10251-102/J10251-102_TESS_TIC36879392_S09.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10238+438/J10238+438_TESS_TIC392365135_S21.fits filename DOES exist


Started querying MAST for observations with the exact target_name='095431305'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC95431305'.


nan filename DOES NOT exist, trying to look and download


MAST found 7 observations. Now querying MAST for the corresponding data products.
MAST found 0 matching data products.


** WARNING: no results found for J10196+198 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J10185-117/J10185-117_TESS_TIC36724087_S09.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10182-204/J10182-204_TESS_TIC192543856_S09.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10167-119/J10167-119_TESS_TIC385194991_S09.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10125+570/J10125+570_TESS_TIC137958028_S21.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10122-037/J10122-037_TESS_TIC77612635_S08.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10088+692/J10088+692_TESS_TIC103633434_S14.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J10023+480/J10023+480_TESS_TIC252470070_S21.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09597+472/J09597+472_TESS_TIC157835914_S21.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09561+627/J09561+627_TESS_TIC28

Started querying MAST for observations with the exact target_name='307646256'.
MAST found 4 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J09411+132 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J09360-216/J09360-216_TESS_TIC413248763_S08.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09307+003/J09307+003_TESS_TIC383218397_S08.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09286-121/J09286-121_TESS_TIC296739893_S35.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09163-186/J09163-186_TESS_TIC5362024_S08.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09161+018/J09161+018_TESS_TIC290474796_S08.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09144+526/J09144+526_TESS_TIC251078597_S21.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09143+526/J09143+526_TESS_TIC251078595_S21.fits filename DOES exist


Started querying MAST for observations with the exact target_name='085993728'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC85993728'.


nan filename DOES NOT exist, trying to look and download


MAST found 3 observations. Now querying MAST for the corresponding data products.
MAST found 0 matching data products.


** WARNING: no results found for J09140+196 star


Started querying MAST for observations with the exact target_name='802602073'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 2 matching data products.
File found in local cache.


*** SOME ERROR happened. Error = object has no attribute centroid_col
../data/CARMENES_GTO_TESS_lc/lightcurves/J09033+056/J09033+056_TESS_TIC270511002_S08.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09028+680/J09028+680_TESS_TIC219704346_S14.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J09005+465/J09005+465_TESS_TIC29168887_S21.fits filename DOES exist


Started querying MAST for observations with the exact target_name='203225829'.


nan filename DOES NOT exist, trying to look and download


MAST found 7 observations. Now querying MAST for the corresponding data products.
MAST found 0 matching data products.


** WARNING: no results found for J09003+218 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J08599+729/J08599+729_TESS_TIC207339173_S14.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08536-034/J08536-034_TESS_TIC7975441_S08.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08526+283/J08526+283_TESS_TIC332064684_S21.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08413+594/J08413+594_TESS_TIC445064836_S20.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08409-234/J08409-234_TESS_TIC118608254_S08.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08402+314/J08402+314_TESS_TIC341526939_S21.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08358+680/J08358+680_TESS_TIC102885744_S20.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08315+730/J08315+730_TESS_TIC312885008_S20.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08298+267/J08298+267_TESS_TIC36

Started querying MAST for observations with the exact target_name='455139555'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC455139555'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC455139555".


** WARNING: no results found for J08161+013 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J08126-215/J08126-215_TESS_TIC437127546_S07.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08119+087/J08119+087_TESS_TIC334415465_S07.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J08023+033/J08023+033_TESS_TIC452866790_S07.fits filename DOES exist


Started querying MAST for observations with the exact target_name='019028197'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC19028197'.


nan filename DOES NOT exist, trying to look and download


MAST found 7 observations. Now querying MAST for the corresponding data products.
MAST found 0 matching data products.


** WARNING: no results found for J07590+153 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J07582+413/J07582+413_TESS_TIC371272665_S20.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07558+833/J07558+833_TESS_TIC289972535_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07472+503/J07472+503_TESS_TIC309661100_S20.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07446+035/J07446+035_TESS_TIC266744225_S07.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07403-174/J07403-174_TESS_TIC289113766_S07.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07393+021/J07393+021_TESS_TIC280313260_S07.fits filename DOES exist


Started querying MAST for observations with the exact target_name='415085843'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC415085843'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC415085843".


** WARNING: no results found for J07386-212 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J07361-031/J07361-031_TESS_TIC65673065_S07.fits filename DOES exist


Started querying MAST for observations with the exact target_name='355061271'.
MAST found 1 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J07353+548 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J07319+362N/J07319+362N_TESS_TIC16034007_S20.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07287-032/J07287-032_TESS_TIC64910894_S07.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07274+052/J07274+052_TESS_TIC318686860_S07.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07051-101/J07051-101_TESS_TIC177122224_S07.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07044+682/J07044+682_TESS_TIC417758604_S20.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07033+346/J07033+346_TESS_TIC165204611_S20.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J07001-190/J07001-190_TESS_TIC80033631_S06.fits filename DOES exist


Started querying MAST for observations with the exact target_name='058998044'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC58998044'.


nan filename DOES NOT exist, trying to look and download


MAST found 4 observations. Now querying MAST for the corresponding data products.
MAST found 0 matching data products.


** WARNING: no results found for J06594+193 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J06574+740/J06574+740_TESS_TIC141476700_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J06548+332/J06548+332_TESS_TIC68581262_S20.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J06421+035/J06421+035_TESS_TIC233975043_S06.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J06396-210/J06396-210_TESS_TIC48523939_S06.fits filename DOES exist


Started querying MAST for observations with the exact target_name='307809773'.
MAST found 4 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J06371+175 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J06318+414/J06318+414_TESS_TIC144282456_S20.fits filename DOES exist


Started querying MAST for observations with the exact target_name='430044614'.
MAST found 6 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J06246+234 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J06105-218/J06105-218_TESS_TIC124279525_S06.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J06103+821/J06103+821_TESS_TIC138421058_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J06024+498/J06024+498_TESS_TIC467119250_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J06011+595/J06011+595_TESS_TIC322347050_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J06000+027/J06000+027_TESS_TIC282501711_S06.fits filename DOES exist


Started querying MAST for observations with the exact target_name='720817989'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC720817989'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC720817989".


** WARNING: no results found for J05532+242 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J05421+124/J05421+124_TESS_TIC247463344_S06.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05415+534/J05415+534_TESS_TIC311064351_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05394+406/J05394+406_TESS_TIC239097694_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05366+112/J05366+112_TESS_TIC436248798_S06.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05365+113/J05365+113_TESS_TIC436248822_S06.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05360-076/J05360-076_TESS_TIC388934726_S06.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05348+138/J05348+138_TESS_TIC52373734_S06.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05337+019/J05337+019_TESS_TIC220044948_S06.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05314-036/J05314-036_TESS_TIC5

Started querying MAST for observations with the exact target_name='467611768'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J05127+196 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J05084-210/J05084-210_TESS_TIC146539195_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05062+046/J05062+046_TESS_TIC455029978_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05033-173/J05033-173_TESS_TIC169304560_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05019-069/J05019-069_TESS_TIC43605290_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J05019+011/J05019+011_TESS_TIC269077806_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04588+498/J04588+498_TESS_TIC354432728_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04538-177/J04538-177_TESS_TIC246902275_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04520+064/J04520+064_TESS_TIC450102450_S05.fits filename DOES exist


Started querying MAST for observations with the exact target_name='238413768'.


nan filename DOES NOT exist, trying to look and download


MAST found 2 observations. Now querying MAST for the corresponding data products.
MAST found 0 matching data products.


** WARNING: no results found for J04472+206 star


Started querying MAST for observations with the exact target_name='397356212'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J04429+214 star


Started querying MAST for observations with the exact target_name='397354290'.
MAST found 4 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J04429+189 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J04406-128/J04406-128_TESS_TIC117799904_S32.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04376-110/J04376-110_TESS_TIC55997717_S32.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04376+528/J04376+528_TESS_TIC274545692_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04343+430/J04343+430_TESS_TIC28900646_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04311+589/J04311+589_TESS_TIC356322721_S19.fits filename DOES exist


Started querying MAST for observations with the exact target_name='017558287'.
No observations found. Now performing a cone search instead.


nan filename DOES NOT exist, trying to look and download


Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC17558287'.
MAST found 2 observations. Now querying MAST for the corresponding data products.
MAST found 0 matching data products.


** WARNING: no results found for J04290+219 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J04225+105/J04225+105_TESS_TIC372896796_S32.fits filename DOES exist


Started querying MAST for observations with the exact target_name='017307715'.
No observations found. Now performing a cone search instead.


nan filename DOES NOT exist, trying to look and download


Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC17307715'.
MAST found 5 observations. Now querying MAST for the corresponding data products.
MAST found 0 matching data products.


** WARNING: no results found for J04219+213 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J04198+425/J04198+425_TESS_TIC347994537_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04173+088/J04173+088_TESS_TIC283866910_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04167-120/J04167-120_TESS_TIC70899085_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J04153-076/J04153-076_TESS_TIC671834302_S32.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J03531+625/J03531+625_TESS_TIC393080288_S19.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J03473-019/J03473-019_TESS_TIC333680372_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J03473+086/J03473+086_TESS_TIC468889408_S05.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J03463+262/J03463+262_TESS_TIC125736050_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J03230+420/J03230+420_TESS_TIC3

Started querying MAST for observations with the exact target_name='357936741'.
No observations found. Now performing a cone search instead.


nan filename DOES NOT exist, trying to look and download


Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC357936741'.
MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC357936741".


** WARNING: no results found for J03213+799 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J03181+382/J03181+382_TESS_TIC178078333_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J03142+286/J03142+286_TESS_TIC35025268_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J03133+047/J03133+047_TESS_TIC328465904_S04.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J03090+100/J03090+100_TESS_TIC365369408_S31.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J02573+765/J02573+765_TESS_TIC396562848_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J02565+554W/J02565+554W_TESS_TIC251281112_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J02560-006/J02560-006_TESS_TIC439874904_S31.fits filename DOES exist


Started querying MAST for observations with the exact target_name='257870150'.
MAST found 1 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J02530+168 star


Started querying MAST for observations with the exact target_name='371422484'.
MAST found 3 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J02519+224 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J02489-145W/J02489-145W_TESS_TIC29960110_S04.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J02489-145E/J02489-145E_TESS_TIC29960109_S04.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J02486+621/J02486+621_TESS_TIC51024887_S18.fits filename DOES exist


Started querying MAST for observations with the exact target_name='218088471'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC218088471'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC218088471".


** WARNING: no results found for J02465+164 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J02442+255/J02442+255_TESS_TIC436856183_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J02362+068/J02362+068_TESS_TIC278962913_S31.fits filename DOES exist


Started querying MAST for observations with the exact target_name='247042708'.
MAST found 3 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J02358+202 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J02336+249/J02336+249_TESS_TIC22750743_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J02222+478/J02222+478_TESS_TIC409527090_S18.fits filename DOES exist


Started querying MAST for observations with the exact target_name='408587159'.
MAST found 4 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J02164+135 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J02123+035/J02123+035_TESS_TIC270415707_S04.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J02088+494/J02088+494_TESS_TIC250602194_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J02070+496/J02070+496_TESS_TIC250559611_S18.fits filename DOES exist


Started querying MAST for observations with the exact target_name='347509900'.
MAST found 4 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J02022+103 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J02015+637/J02015+637_TESS_TIC374029463_S18.fits filename DOES exist


Started querying MAST for observations with the exact target_name='404715018'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC404715018'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC404715018".


** WARNING: no results found for J02002+130 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J01550+379/J01550+379_TESS_TIC67390998_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J01518+644/J01518+644_TESS_TIC10352491_S18.fits filename DOES exist


Started querying MAST for observations with the exact target_name='369534824'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J01433+043 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J01352-072/J01352-072_TESS_TIC29853348_S03.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J01339-176/J01339-176_TESS_TIC54003038_S03.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J01125-169/J01125-169_TESS_TIC439403362_S03.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J01066+192/J01066+192_TESS_TIC243185500_S17.fits filename DOES exist


Started querying MAST for observations with the exact target_name='015615674'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC15615674'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC15615674".


** WARNING: no results found for J01056+284 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J01048-181/J01048-181_TESS_TIC268633473_S03.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J01033+623/J01033+623_TESS_TIC52183206_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J01026+623/J01026+623_TESS_TIC256419669_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J01025+716/J01025+716_TESS_TIC379084450_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J01019+541/J01019+541_TESS_TIC252110114_S17.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J01013+613/J01013+613_TESS_TIC256316623_S18.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J00570+450/J00570+450_TESS_TIC196293610_S17.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J00403+612/J00403+612_TESS_TIC284441182_S17.fits filename DOES exist
../data/CARMENES_GTO_TESS_lc/lightcurves/J00389+306/J00389+306_TESS_TIC2

Started querying MAST for observations with the exact target_name='440109718'.
MAST found 1 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 1 matching data products.
File found in local cache.


*** SOME ERROR happened. Error = object has no attribute centroid_col
../data/CARMENES_GTO_TESS_lc/lightcurves/J00183+440/J00183+440_TESS_TIC440109725_S17.fits filename DOES exist


Started querying MAST for observations with the exact target_name='150899679'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC150899679'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC150899679".


** WARNING: no results found for J00162+198W star


Started querying MAST for observations with the exact target_name='150899677'.
No observations found. Now performing a cone search instead.
Started querying MAST for observations within 0.0001 arcsec arcsec of objectname='TIC150899677'.


nan filename DOES NOT exist, trying to look and download


MAST found 0 observations. Now querying MAST for the corresponding data products.
No data found for target "TIC150899677".


** WARNING: no results found for J00162+198E star


Started querying MAST for observations with the exact target_name='176287658'.
MAST found 2 observations. Now querying MAST for the corresponding data products.


nan filename DOES NOT exist, trying to look and download


MAST found 0 matching data products.


** WARNING: no results found for J00067-075 star
../data/CARMENES_GTO_TESS_lc/lightcurves/J00051+457/J00051+457_TESS_TIC439946126_S17.fits filename DOES exist


In [9]:
i

389

### Check final numbers:

In [10]:
gto[gto['lc_file'] == "<ambiguous>"]

Empty DataFrame
Columns: [Karmn, Name, Comp, GJ, RA_J2016_deg, DE_J2016_deg, RA_J2000, DE_J2000, l_J2016_deg, b_J2016_deg, Ref01, SpT, SpTnum, Ref02, Teff_K, eTeff_K, logg, elogg, [Fe/H], e[Fe/H], Ref03, L_Lsol, eL_Lsol, Ref04, R_Rsol, eR_Rsol, Ref05, M_Msol, eM_Msol, Ref06, muRA_masa-1, emuRA_masa-1, muDE_masa-1, emuDE_masa-1, Ref07, pi_mas, epi_mas, Ref08, d_pc, ed_pc, Ref09, Vr_kms-1, eVr_kms-1, Ref10, ruwe, Ref11, U_kms-1, eU_kms-1, V_kms-1, eV_kms-1, W_kms-1, eW_kms-1, Ref12, sa_m/s/a, esa_m/s/a, Ref13, SKG, Ref14, SKG_lit, Ref14_lit, Pop, Ref15, vsini_flag, vsini_kms-1, evsini_kms-1, Ref16, P_d, eP_d, Ref17, pEWHalpha_A, epEWHalpha_A, Ref18, log(LHalpha/Lbol), elog(LHalpha/Lbol), Ref19, 1RXS, CRT_s-1, eCRT_s-1, HR1, eHR1, HR2, eHR2, Flux_X_E-13_ergcm-2s-1, eFlux_X_E-13_ergcm-2s-1, LX/LJ, eLX/LJ, Ref20, Activity, Ref21, FUV_mag, eFUV_mag, NUV_mag, eNUV_mag, Ref22, u_mag, eu_mag, Ref23, BT_mag, eBT_mag, Ref24, ...]
Index: []

[0 rows x 248 columns]

In [11]:
gto[gto['has_lc'] == False][['has_lc', 'lc_file']]

has_lc lc_file
0     False     NaN
1     False     NaN
4     False     NaN
6     False     NaN
8     False     NaN
..      ...     ...
373   False     NaN
384   False     NaN
386   False     NaN
387   False     NaN
388   False     NaN

[119 rows x 2 columns]

## Store the result

In [12]:
gto.to_csv(DATA_FILE, sep=',', decimal='.', index=False)

**OBSERVATIONS AND CONCLUSIONS:**
- We have tried to complete the TESS light curves.
- Noticed that downloaded files were much smaller than the existing ones, so later decided to just keep the existing ones, and only try to downlñoad if no file existed previously.
- Finally, 110 objects (about one third of the total) remained with no available TESS light curve.